src/ResourceManager/Kusto/Commands.Kusto/Commands/UpdateAzureRmKustoCluster.cs

```diff
@@ -0,0 +1,152 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Management.Automation;
+using System.Security.Permissions;
+using Microsoft.Azure.Commands.Kusto.Models;
+using Microsoft.Azure.Commands.Kusto.Properties;
+using Microsoft.Azure.Commands.Kusto.Utilities;
+using Microsoft.Rest.Azure;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.Kusto.Commands
+{
+
+    [Cmdlet("Update", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "KustoCluster",
+         SupportsShouldProcess = true),
+     OutputType(typeof(PSKustoCluster))]
+    public class UpdateAzureRmKustoCluster : KustoCmdletBase
+    {
+        protected const string CmdletParametersSet = "ByNameAndResourceGroup";
+        protected const string ObjectParameterSet = "ByInputObject";
+        protected const string ResourceIdParameterSet = "ByResourceId";
+
+        [Parameter(
+            ParameterSetName = CmdletParametersSet,
+            Mandatory = true,
+            HelpMessage = "Name of cluster to be updated.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            ParameterSetName = CmdletParametersSet,
+            Mandatory = false,
+            HelpMessage = "Name of resource group under which the cluster exists.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = CmdletParametersSet,
+            Mandatory = true,
+            HelpMessage = "Azure region where the cluster exists.")]
+        [LocationCompleter("Microsoft.Kusto/clusters")]
+        public string Location { get; set; }
```

Comment:
do you need this parameter? It looks like you don't use it in the "Get", and you could just grab it from the object received from the get in order to send it to the update call.  It is my understanding that you cannot change the location so it shouldn't be included for that reason either.

Reply:
I removed the line that takes the location from the cluster.
Currently, we are not allowing clusters in different regions to have the same name, but that might change in the future